File: 392.is-subsequence.py
#
# @lc app=leetcode id=392 lang=python
#
# [392] Is Subsequence
#

# @lc code=start
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        lens = len(s)
        lent = len(t)
        i = 0
        j = 0
        if (lens == 0):
            return True
        if lent == 0:
            return False
        while j < lent:
            if i == lens:
                return True
            if s[i] == t[j]:
                i += 1
                j += 1
            else:
                j += 1
        return i == lens
        
# @lc code=end

File: 392.is-subsequence.java
/*
 * @lc app=leetcode id=392 lang=java
 *
 * [392] Is Subsequence
 */

// @lc code=start
class Solution {
    public boolean isSubsequence(String s, String t) {
        int lens = s.length();
        int lent = t.length();
        int i = 0;
        int j = 0;
        if (lens == 0) {
            return true;
        }
        if (lent == 0) {
            return false;
        }
        while (j < lent) {
            if (i == lens) {
                return true;
            }
            if (s.charAt(i) == t.charAt(j)) {
                i++;
                j++;
            }
            else {
                j++;
            }
        }
        return (i == lens);
    }
}
// @lc code=end

